kin-devplatform-sdk/src/main/java/kin/devplatform/base/KinEcosystemBaseActivity.java

```diff
@@ -0,0 +1,21 @@
+package kin.devplatform.base;
+
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import kin.devplatform.KinEcosystemInitiator;
+
+public abstract class KinEcosystemBaseActivity extends AppCompatActivity {
+
+	protected abstract @LayoutRes
+	int getLayoutRes();
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(getLayoutRes());
+
+		KinEcosystemInitiator.getInstance().internalInit(this);
```

Comment:
Just want to make sure, are we positive that this won't happen in the first time that the app is starting?
Meaning that we will accidentally call internalInit for the first time before we called to external init?

Reply:
it won't happen until developer called `start` and launch marketplace, as launch marketplace is blocked until start is finished. so `externalInit` must be called at least once and finish successfully before some ecosystem activity will be launched.